Sources/SwiftGenKit/Parsers/ParserOptionList.swift

```diff
@@ -0,0 +1,44 @@
+//
+// SwiftGenKit
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+public struct ParserOptionList {
+  enum Error: Swift.Error {
+    case unknownOption(key: String, value: Any)
+  }
+
+  private let options: [AnyParserOption]
+
+  init(options: [AnyParserOption] = []) {
+    self.options = options
+  }
```

Comment:
Bonus points: make `ParserOptionList: ExpressibleByArrayLiteral`.

That way we'd then be able to write:

```swift
static let allOptions: ParserOptionList = [
  Option.separator
]
```
And if we end up having more and more options, this would make it align and read nicely 😜